### ISSUE TITLE
Changelog file destination must be inside of the repository directory.

### ISSUE THREAD

**Issue by rantoniuk** [python-semantic-release/python-semantic-release#978]
On version 9.8.0, the following command works:
```
APP_VER=$(semantic-release -c .semantic-release.toml version \
  --no-push --no-tag --no-commit --no-changelog --no-vcs-release --print \
  --build-metadata pr-${{ github.event.pull_request.number }}.${{ github.run_number }})
```

with the output:

```
Installing collected packages: zipp, typing-extensions, tomlkit, smmap, shellingham, requests, pygments, mdurl, MarkupSafe, dotty-dict, colorama, requests-toolbelt, pydantic-core, markdown-it-py, jinja2, importlib-resources, gitdb, click, annotated-types, rich, python-gitlab, pydantic, gitpython, click-option-group, python-semantic-release
  Attempting uninstall: requests
    Found existing installation: requests 2.31.0
    Uninstalling requests-2.31.0:
      Successfully uninstalled requests-2.31.0
Successfully installed MarkupSafe-2.1.5 annotated-types-0.7.0 click-8.1.7 click-option-group-0.5.6 colorama-0.4.6 dotty-dict-1.3.1 gitdb-4.0.11 gitpython-3.1.43 importlib-resources-6.4.0 jinja2-3.1.4 markdown-it-py-3.0.0 mdurl-0.1.2 pydantic-2.8.2 pydantic-core-2.20.1 pygments-2.18.0 python-gitlab-4.7.0 python-semantic-release-9.8.0 requests-2.32.3 requests-toolbelt-1.0.0 rich-13.7.1 shellingham-1.5.4 smmap-5.0.1 tomlkit-0.13.0 typing-extensions-4.12.2 zipp-3.19.2
Switched to a new branch '267/merge'
[09:37:09] WARNING  [semantic_release.cli.config] WARNING         config.py:523
                    config.from_raw_config: Token value is                     
                    missing!                                                   
[09:37:10] WARNING  [semantic_release.cli.config] WARNING         config.py:523
                    config.from_raw_config: Token value is                     
                    missing!                                                   
APP_VER=1.26.0+pr-267.211
```

On 9.8.5:
```
Installing collected packages: zipp, typing-extensions, tomlkit, smmap, shellingham, requests, pygments, mdurl, MarkupSafe, dotty-dict, colorama, requests-toolbelt, pydantic-core, markdown-it-py, jinja2, importlib-resources, gitdb, click, annotated-types, rich, python-gitlab, pydantic, gitpython, click-option-group, python-semantic-release
  Attempting uninstall: requests
    Found existing installation: requests 2.31.0
    Uninstalling requests-2.31.0:
      Successfully uninstalled requests-2.31.0
Successfully installed MarkupSafe-2.1.5 annotated-types-0.7.0 click-8.1.7 click-option-group-0.5.6 colorama-0.4.6 dotty-dict-1.3.1 gitdb-4.0.11 gitpython-3.1.43 importlib-resources-6.4.0 jinja2-3.1.4 markdown-it-py-3.0.0 mdurl-0.1.2 pydantic-2.8.2 pydantic-core-2.20.1 pygments-2.18.0 python-gitlab-4.7.0 python-semantic-release-9.8.5 requests-2.32.3 requests-toolbelt-1.0.0 rich-13.7.1 shellingham-1.5.4 smmap-5.0.1 tomlkit-0.13.0 typing-extensions-4.12.2 zipp-3.19.2
Switched to a new branch '267/merge'
[09:15:57] WARNING  [semantic_release.cli.config] WARNING         config.py:558
                    config.from_raw_config: Token value is                     
                    missing!                                                   
Changelog file destination must be inside of the repository directory.
##[error]Process completed with exit code 1.
```

Interestingly, I cannot reproduce this locally, only on GH actions.
Pinning the version to 9.8.0 resolved the issue for me.

**Comment by codejedi365** [comment]
Can you provide your configuration file contents? 

That error is related to the config option `semantic_release.changelog.changelog_file` value. Unless my path resolution is incorrect, I added the restriction to prevent writing changelogs outside of the repository directory.

Unfortunately the configuration is validated separately and regardless of the `--no-changelog` flag. 

Also you should be fine up to and including `v9.8.3`. `v9.8.4` introduced the changelog destination validation.

**Comment by rantoniuk** [comment]
```
$ cat .semantic-release.toml

[semantic_release]
assets = []
commit_message = "{version}\n\nauto-version by python-semantic-release"
commit_parser = "angular"
logging_use_named_masks = false
major_on_zero = true
tag_format = "v{version}"

[semantic_release.branches.issue]
match = ".*"
prerelease_token = "rc"
prerelease = false
```

**Comment by codejedi365** [comment]
Ok, thanks! 

It looks like somehow the default value is failing since you are not overriding it. 

Can you provide your GitHub actions job configuration? 

It has to do with what it perceives as the current working directory when resolving the changelog file path (where it would write the changelog to). The  default changelog is a relative path to the current working directory. The check verifies that the destination file will be inside any of the child directories of the git repository.

**Comment by rantoniuk** [comment]
My action configuration is too complex to show, but I would guess this might somehow be related to either:
- not providing the token (this is anyway noop so should not be the case)
- or doing `git switch -c $GITHUB_REF`

However, I was not able to reproduce it locally on my mac.

Part of the workflow definition is:

```
jobs:
  build-app:
      runs-on: Windows-8CPU-32GB

      steps:
        - name: "Github Actions Auth"
          id: ghactions-auth
          uses: actions/create-github-app-token@v1
          with:
            app-id: ${{ vars.GHACTIONS_PAT_APP_ID }}
            private-key: ${{ secrets.GHACTIONS_PAT_APP_ID_PK }}
            owner: ${{ github.repository_owner }}

        - name: "Checkout"
          uses: actions/checkout@v4
          with:
            token: ${{ steps.ghactions-auth.outputs.token }}
            fetch-depth: 0

...

       - name: Calculate version
          id: version
          shell: bash
          run: |
            echo "REF: $GITHUB_REF"
            echo "REF_NAME: $GITHUB_REF_NAME"

            pip install python-semantic-release

            if [[ "${{ github.event_name }}" == "pull_request" ]]; then
              git switch -c $GITHUB_REF_NAME
              APP_VER=$(semantic-release -c .semantic-release.toml version \
                --no-push --no-tag --no-commit --no-changelog --no-vcs-release --print \
                --build-metadata pr-${{ github.event.pull_request.number }}.${{ github.run_number }})
              INNO_FILE_VER=$(semantic-release -c .semantic-release.toml version \
                --no-push --no-tag --no-commit --no-changelog --no-vcs-release --print)
            else
              git switch -c $GITHUB_REF
              APP_VER=$(semantic-release -c .semantic-release.toml version \
                --no-push --no-tag --no-commit --no-changelog --no-vcs-release --print \
                --build-metadata build.${{ github.run_number }})
            fi

```

Note it is running on **Windows**, not on Linux ;-)

**Comment by codejedi365** [comment]
Ok, it is almost certainly windows and its path resolution results.

Thank you for providing that snippet. I thought you were using our GitHub Action which runs in a Linux container at first.

**Comment by codejedi365** [comment]
### :tada: This resolution has been included in version 9.8.8 :tada:

The release is available on:
- [PyPi package](https://pypi.org/project/python-semantic-release/9.8.8/)
- [GitHub release](https://github.com/python-semantic-release/python-semantic-release/releases/tag/v9.8.8)
